src/luacov/linescanner.lua

```diff
@@ -39,7 +39,7 @@ end
 --- Lines that are always excluded from accounting
 local any_hits_exclusions = {
    "", -- Empty line
-   "end[,; %)]*", -- Single "end"
+   "end[,; %)]*[end[,; %)]*]*", -- One or more "end"s
```

Comment:
These are Lua patterns and not regular expressions; they do not allow for nested rules or multi-character sequences within repeats. The rule as you wrote (_if_ it parses as a valid pattern) would catch other things, including the word `ended`, for example.

Reply:
It does parse as a valid pattern.  I'm using it for myself with no problems.

It's not really a matter of personal coding style.  In theory, luacov should be able to correctly identify for any given line of code whether it contains a statement or not.  The line should show up as missing only if it does contain a statement and no statement on that line ever executes.  It should be possible to make that determination using nothing other than knowledge of lua syntax, with no reference to style whatsoever.

This is only a few characters' worth of change, and it gets luacov a tiny bit closer to that.

But I do get that, as it stands, luacov does make assumptions about style, and then there's a question of where to draw the line.